shared/actions/provision.js

```diff
@@ -485,6 +485,18 @@ const showFinalErrorPage = (state, action) => {
 const showUsernameEmailPage = () =>
   RouteTreeGen.createNavigateAppend({parentPath: [Tabs.loginTab], path: ['usernameOrEmail']})
 
+const forgotUsername = (state, action) =>
+  RPCTypes.accountRecoverUsernameRpcPromise({email: action.payload.email}, Constants.forgotUsernameWaitingKey)
+    .then(result => ProvisionGen.createForgotUsernameResult({result: 'success'}))
+    .catch(error =>
+      ProvisionGen.createForgotUsernameResult({
+        result:
+          error.code === RPCTypes.constantsStatusCode.scnotfound
+            ? "We couldn't find an account with that email address. Try again?"
+            : error.desc,
```

Comment:
do we also want to rewrite "bad email address" error message?  I can make the service return other status code rather than INPUT_ERROR (100) as well

Reply:
Thanks, done! That error code's fine.